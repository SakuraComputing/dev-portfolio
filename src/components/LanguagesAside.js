import React from 'react';

export const LanguagesAside = () => (
    <aside className="languages aside section">
        <div className="section-inner">
            <h2 className="heading">Languages</h2>
            <div className="content">
                <ul className="list-unstyled">
                    <li className="item">
                        <span className="title"><strong>English:</strong></span>
                        <span className="level">Native Speaker <br className="visible-xs"/><i
                            className="fa fa-star"></i> <i className="fa fa-star"></i> <i className="fa fa-star"></i> <i
                            className="fa fa-star"></i> <i className="fa fa-star"></i> </span>
                    </li>
                    <li className="item">
                        <span className="title"><strong>Japanese:</strong></span>
                        <span className="level">Professional Proficiency <br className="visible-sm visible-xs"/><i
                            className="fa fa-star"></i> <i className="fa fa-star"></i></span>
                    </li>
                </ul>
            </div>
        </div>
    </aside>

);
export default LanguagesAside;