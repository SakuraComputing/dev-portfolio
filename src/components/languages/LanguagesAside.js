import React from 'react';
import Language from './Language';

class LanguagesAside extends React.Component {
    render() {

        const { languages } = this.props;

        let languageItem;

        if(languages.length > 0) {
            languageItem = languages.map(language => (
                <Language key={language.title} language={language}/>
                )
            )
        }


        return (
            <aside className="languages aside section">
                <div className="section-inner">
                    <h2 className="heading">Languages</h2>
                    <div className="content">
                        <ul className="list-unstyled">
                            {languageItem}
                        </ul>
                    </div>
                </div>
            </aside>
        )
    }
}
export default LanguagesAside;
