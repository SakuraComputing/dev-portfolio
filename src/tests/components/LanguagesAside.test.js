import React from 'react';
import { shallow } from 'enzyme';
import LanguagesAside from '../../components/LanguagesAside';

describe('LanguagesAside Section', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<LanguagesAside />);
        expect(wrapper).toMatchSnapshot();
    });
});