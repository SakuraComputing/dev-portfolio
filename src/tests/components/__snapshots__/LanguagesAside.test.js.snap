// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguagesAside Section should render correctly 1`] = `
<aside
  className="languages aside section"
>
  <div
    className="section-inner"
  >
    <h2
      className="heading"
    >
      Languages
    </h2>
    <div
      className="content"
    >
      <ul
        className="list-unstyled"
      >
        <li
          className="item"
        >
          <span
            className="title"
          >
            <strong>
              English:
            </strong>
          </span>
          <span
            className="level"
          >
            Native Speaker 
            <br
              className="visible-xs"
            />
            <i
              className="fa fa-star"
            />
             
            <i
              className="fa fa-star"
            />
             
            <i
              className="fa fa-star"
            />
             
            <i
              className="fa fa-star"
            />
             
            <i
              className="fa fa-star"
            />
             
          </span>
        </li>
        <li
          className="item"
        >
          <span
            className="title"
          >
            <strong>
              Spanish:
            </strong>
          </span>
          <span
            className="level"
          >
            Professional Proficiency 
            <br
              className="visible-sm visible-xs"
            />
            <i
              className="fa fa-star"
            />
             
            <i
              className="fa fa-star"
            />
             
            <i
              className="fa fa-star"
            />
             
            <i
              className="fa fa-star-half"
            />
          </span>
        </li>
      </ul>
    </div>
  </div>
</aside>
`;
